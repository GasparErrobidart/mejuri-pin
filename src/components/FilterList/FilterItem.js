import React                from 'react'
import StyledFilterItem     from './StyledFilterItem'
import FilterItemWrapper    from './FilterItemWrapper'
import { FiCheck }          from 'react-icons/fi';

const FilterItem = (props)=>{
  const {children,checked} = props
  let icon = checked ? <FiCheck/> : null
  return(
    <StyledFilterItem {...props}>
      <FilterItemWrapper {...props}>
          <div>
            {icon}
          </div>
          <div>
            {children}
          </div>
      </FilterItemWrapper>
    </StyledFilterItem>
  )
}

export default FilterItem
