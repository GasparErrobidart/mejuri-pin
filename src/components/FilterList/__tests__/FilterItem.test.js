import '@testing-library/jest-dom'
import 'jest-styled-components'

import React                 from 'react'
import {ThemeProvider}       from 'styled-components'
import { cleanup }           from '@testing-library/react'
import renderer              from 'react-test-renderer'

import theme                 from '../../GlobalStyle/themes/mejuri-pin'

import FilterItem   from '../FilterItem'


afterEach(cleanup)

test('Renders "FilterItem" component', () => {
  const testText        = "FilterItem Test Text"
  const component       = renderer.create(
    <ThemeProvider theme={theme}>
      <FilterItem>{testText}</FilterItem>
    </ThemeProvider>);
  const tree            = component.toJSON()
  expect(tree).toMatchSnapshot();
});
