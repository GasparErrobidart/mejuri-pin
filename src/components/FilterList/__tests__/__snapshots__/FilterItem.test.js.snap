// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders "FilterItem" component 1`] = `
.c0 {
  background: white;
  padding: 15px 0;
  border-top: 1px solid #F3F3F3;
  cursor: default;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 30px;
  cursor: pointer;
}

.c1:hover,
.c1:active {
  border-color: #CCCCCC;
}

.c1 * {
  -webkit-transition: all 200ms ease-in;
  transition: all 200ms ease-in;
  color: #888888;
}

.c1:hover * {
  color: #303030;
}

.c1 div:first-child {
  overflow: hidden;
  -webkit-flex: 0 0 0;
  -ms-flex: 0 0 0;
  flex: 0 0 0;
  height: 15px;
}

@media (min-width:768px) {
  .c0 {
    background: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div />
    <div>
      FilterItem Test Text
    </div>
  </div>
</div>
`;
